pkg/auth/token_verifier.go

```diff
@@ -123,22 +126,31 @@ func (v *OIDCTokenVerifier) verifyAuthN(ctx context.Context, audience *string, r
 }
 
 // verifyAuthZ verifies if the given idToken is allowed by the resources eventPolicyStatus
-func (v *OIDCTokenVerifier) verifyAuthZ(features feature.Flags, idToken *IDToken, resourceNamespace string, policyRefs []duckv1.AppliedEventPolicyRef, resp http.ResponseWriter) error {
+func (v *OIDCTokenVerifier) verifyAuthZ(ctx context.Context, features feature.Flags, idToken *IDToken, resourceNamespace string, policyRefs []duckv1.AppliedEventPolicyRef, req *http.Request, resp http.ResponseWriter) error {
 	if len(policyRefs) > 0 {
-		subjectsFromApplyingPolicies := []string{}
+		message := cehttp.NewMessageFromHttpRequest(req)
+		defer message.Finish(nil)
+
+		event, err := binding.ToEvent(ctx, message)
+		if err != nil {
+			resp.WriteHeader(http.StatusInternalServerError)
+			return fmt.Errorf("failed to decode event from request: %w", err)
+		}
```

Comment:
Not entirely sure about the calling stack here, but won't we run into issues, when we read the event/message here from the request? IIRC in the brokers ingress_handler (which uses the tokenverifiyer), we read the event from the request too.

Reply:
hmm yeah good point, let me try and fix that...